app/core/pokemon-entity.ts

```diff
@@ -1547,7 +1569,7 @@ export class PokemonEntity extends Schema implements IPokemonEntity {
 
   // called after death (does not proc if resurection)
   onDeath({ board }: { board: Board }) {
-    this.refToBoardPokemon.deathCount++
+    this.addPermanentStats("deathCount", 1)
```

Comment:
confusing method name here, this is not a stat. I'd rather keep original code with a if condition for ghosts